### PR TITLE
refactor(types): update resource types.go and autogenerated code

### PR DESCRIPTION
**Pull request by ksatchit**
Signed-off-by: ksatchit <karthik.s@openebs.io>

**What this PR does / why we need it**:

- Populates _types.go files generated by the operator-sdk with actual resource specs
- Adds autogenerated code upon operator-sdk generate k8s after updates to types.go files
- Minor edits to the resource deploy specs in deploy/crd

**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes #

**Special notes for your reviewer**:

**Checklist:**
- [ ] Fixes #<issue number>
- [ ] Labelled this PR & related issue with `documentation` tag
- [ ] PR messages has document related information
- [ ] Labelled this PR & related issue with `breaking-changes` tag
- [ ] PR messages has breaking changes related information
- [ ] Labelled this PR & related issue with `requires-upgrade` tag
- [ ] PR messages has upgrade related information
- [ ] Commit has unit tests
- [ ] Commit has integration tests

### CODE REVIEW

File: pkg/apis/litmuschaos/v1alpha1/chaosgraph_types.go [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+        // List of ENV vars passed to executor pod
+        Env          []EnvList           `json:"env"`
+        // Defines command to invoke experiment
+        Command	     []string            `json:"command"`
+        // Defines arguments to executor's entrypoint command 
+        Args         []string            `json:"args"`
+}
+
+// EnvList defines env var list to hold chaos params
+type EnvList struct {
```
Comment (AmitKumarDas) [inline pkg/apis/litmuschaos/v1alpha1/chaosgraph_types.go]: Can we rename this to `ENVPair` or just `ENV`

File: pkg/apis/litmuschaos/v1alpha1/chaosgraph_types.go [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+}
+
+// ChaosRunDefinition defines the information fed to the executor framework
+// to enable successful chaos injection
+type ChaosRunDefinition struct {
+        // Default labels of the executor pod
+        // +optional
+        Labels       map[string]string   `json:"labels"`
+        // List of ENV vars passed to executor pod
+        Env          []EnvList           `json:"env"`
```
Comment (AmitKumarDas) [inline pkg/apis/litmuschaos/v1alpha1/chaosgraph_types.go]: Can we rename `Env` to `ENVList`

File: pkg/apis/litmuschaos/v1alpha1/chaosexperiment_types.go [hunk 3]
```diff
@@ -4,15 +4,14 @@ import (
...
-
 // ChaosExperimentSpec defines the desired state of ChaosExperiment
 // +k8s:openapi-gen=true
+// An experiment is the definition of a chaos test and is listed as an item
+// in the chaos engine to be run against a given app.
 type ChaosExperimentSpec struct {
-	// INSERT ADDITIONAL SPEC FIELDS - desired state of cluster
-	// Important: Run "operator-sdk generate k8s" to regenerate code after modifying this file
-	// Add custom validation using kubebuilder tags: https://book.kubebuilder.io/beyond_basics/generating_crd.html
+        // ChaosGraph refers to the resource carrying low-level chaos params
```
Comment (AmitKumarDas) [inline pkg/apis/litmuschaos/v1alpha1/chaosexperiment_types.go]: Can we rewrite the comments as follows:
```go
// ChaosGraph refers to resource carrying low-level
// chaos options
```

File: pkg/apis/litmuschaos/v1alpha1/chaosgraph_types.go [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+        // Defines the chaos parameters and execution artifacts 
+        Definition   ChaosRunDefinition  `json:"definition"`
+}
+
+// ChaosRunDefinition defines the information fed to the executor framework
+// to enable successful chaos injection
+type ChaosRunDefinition struct {
+        // Default labels of the executor pod
+        // +optional
+        Labels       map[string]string   `json:"labels"`
```
Comment (AmitKumarDas) [inline pkg/apis/litmuschaos/v1alpha1/chaosgraph_types.go]: Should we have `Image` as one of the definition options?

File: pkg/apis/litmuschaos/v1alpha1/chaosengine_types.go [hunk 5]
```diff
@@ -4,23 +4,106 @@ import (
...
-	// Add custom validation using kubebuilder tags: https://book.kubebuilder.io/beyond_basics/generating_crd.html
+        //Detailed status of individual experiments
+        Experiments    []ExperimentStatuses   `json:"experiments"`
+}
+
+// ApplicationParams defines information about Application-Under-Test (AUT) on the cluster
+// Controller expects AUT to be annotated with litmuschaos.io/chaos: "true" to run chaos
+type ApplicationParams struct {
+        //Namespace of the AUT
+        Appns          string               `json:"appns"`
```
Comment (chandankumar4) [inline pkg/apis/litmuschaos/v1alpha1/chaosengine_types.go]: What is `Appns` here ?

Comment (ksatchit) [reply pkg/apis/litmuschaos/v1alpha1/chaosengine_types.go]: application's namespace

File: pkg/apis/litmuschaos/v1alpha1/chaosgraph_types.go (hunk 4 above)
Comment (ksatchit) [reply pkg/apis/litmuschaos/v1alpha1/chaosgraph_types.go]: That could definitely be one way... Another would be to delegate the image to the executor infrastructure.. considering that the chaosgraphs are also pulled by charts..

**Pull request merged**